pages/api/newsletter/subscribe.ts

```diff
@@ -0,0 +1,48 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+
+export default async function handler(
+    req: NextApiRequest,
+    res: NextApiResponse
+) {
+    const { email } = req.body
+
+    if (!email) {
+        return res.status(400).json({ error: 'Email is required' })
+    }
+
+    const existingSubscribers = await fetch(
+        'https://www.getrevue.co/api/v2/subscribers',
+        {
+            method: 'GET',
+            headers: {
+                Authorization: `Token ${process.env.REVUE_API_KEY}`,
+                'Content-Type': 'application/json',
+            },
+        }
+    )
+    const subscribers = await existingSubscribers.json()
+
+    if (subscribers.some((sub) => sub.email === email)) {
```

Comment:
This isn't something I feel strongly about, but another thought here could be...
```suggestion
    if (subscribers.some(({ existingEmail }) => existingEmail === email)) {
```

Reply:
If not, then I would probably make `sub` be `subscriber` (singular). Just having `sub` is what is kind of throwing me off.